src/main/java/nl/tudelft/jpacman/game/SinglePlayerGame.java

```diff
@@ -79,5 +107,109 @@ public void moveLeft() {
 	public void moveRight() {
 		move(player, Direction.EAST);
 	}
+	
+	/**
+	 * Override the levelWon method to handle the win of a level
+	 */
+	@Override
+	public void levelWon() {
+		stop();
+		if(levels != null){
+			if(levelIndex + 1 > player.getMaxLevel()){
+				notifyVictory(levelIndex+1);
+				player.setMaxLevel(levelIndex+1);
+			}
+			if(nextLevel()){
+				start();
+			}
+		}
+	}
+	
+	/**
+	 * Override the levelLost method to handle the death of pacman
+	 */
+	@Override
+	public void levelLost() {
+		stop();
+		Random rand = new Random();
+		int x = rand.nextInt(level.getBoard().getWidth());
+		int y = rand.nextInt(level.getBoard().getHeight());
+		if(player.hasLifeRemaining()){
+			Square target = level.getBoard().squareAt(x, y);
+			player.occupy(target.nearestValidRespawn(player, null));
+			player.setAlive(true);
+			start();
+		}else{
+			notifyDefeat();
+		}
+	}
+	
+	/**
+	 * Switches to the next level if the game is in a state accepting it
+	 */
+	public boolean nextLevel() {
+		if((!getLevel().hasBeenStarted() && !isInProgress())
+			|| getLevel().remainingPellets() <= 0){
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://www.codacy.com/app/mathieu-craeye/jpacman-framework_2/file/2535831727/issues/source?bid=3267846&fileBranchId=3284210#l153)